packages/twenty-front/src/modules/ai/components/AIChatAssistantMessageRenderer.tsx

```diff
@@ -0,0 +1,128 @@
+import { ErrorStepRenderer } from '@/ai/components/ErrorStepRenderer';
+import { ReasoningSummaryDisplay } from '@/ai/components/ReasoningSummaryDisplay';
+import { ToolStepRenderer } from '@/ai/components/ToolStepRenderer';
+import type { ParsedStep } from '@/ai/types/streamTypes';
+import { parseStream } from '@/ai/utils/parseStream';
+import { IconDotsVertical } from 'twenty-ui/display';
+
+import { LazyMarkdownRenderer } from '@/ai/components/LazyMarkdownRenderer';
+import { agentStreamingMessageState } from '@/ai/states/agentStreamingMessageState';
+import { keyframes, useTheme } from '@emotion/react';
+import styled from '@emotion/styled';
+import { useRecoilValue } from 'recoil';
+
+const StyledStepsContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: ${({ theme }) => theme.spacing(1)};
+`;
+
+const StyledDotsIconContainer = styled.div`
+  align-items: center;
+  border: ${({ theme }) => `1px solid ${theme.border.color.light}`};
+  border-radius: ${({ theme }) => theme.border.radius.md};
+  display: flex;
+  justify-content: center;
+  padding-inline: ${({ theme }) => theme.spacing(1)};
+`;
+
+const StyledDotsIcon = styled(IconDotsVertical)`
+  color: ${({ theme }) => theme.font.color.light};
+  transform: rotate(90deg);
+`;
+
+const dots = keyframes`
+  0% { content: ''; }
+  33% { content: '.'; }
+  66% { content: '..'; }
+  100% { content: '...'; }
+`;
+
+const StyledToolCallContainer = styled.div`
+  &::after {
+    display: inline-block;
+    content: '';
+    animation: ${dots} 750ms steps(3, end) infinite;
+    width: 2ch;
+    text-align: left;
+  }
+`;
+
+const LoadingDotsIcon = () => {
+  const theme = useTheme();
+
+  return (
+    <StyledDotsIconContainer>
+      <StyledDotsIcon size={theme.icon.size.xl} />
+    </StyledDotsIconContainer>
+  );
+};
+
+export const AIChatAssistantMessageRenderer = ({
+  streamData,
+}: {
+  streamData: string;
+}) => {
+  const agentStreamingMessage = useRecoilValue(agentStreamingMessageState);
+  const isStreaming =
+    Boolean(agentStreamingMessage) && streamData === agentStreamingMessage;
+
+  if (!streamData) {
+    return <LoadingDotsIcon />;
+  }
+
+  const isPlainString =
+    !streamData.includes('\n') ||
+    !streamData.split('\n').some((line) => {
+      try {
+        JSON.parse(line);
+        return true;
+      } catch {
+        return false;
+      }
+    });
+
+  if (isPlainString) {
```

Comment:
Rename to `hasStructuredStreamData` or something more precise. Put in a dedicated file in utils folder in /ai. Simplify double negative condition.

E.G.
```
  const hasStructuredStreamData = (data: string): boolean => {
    if (!data.includes('\n')) {
      return false;
    }
    
    return data.split('\n').some((line) => {
      try {
        JSON.parse(line);
        return true;
      } catch {
        return false;
      }
    });
  };
```